### PR TITLE
fix: Flush transaction handling after replay

### LINKED ISSUE #10923: Flush transaction handling after PCES replay
- Reconnect tests [sometimes fail](http://35.247.76.217:8095/alittley/10841-direct-hashing/4N/Reconnect/20240110-222006-GCP-Daily-Reconnect-4N/VMerkleDual-NDReconnect-500-60m/swirlds-logs.html) when clearing the transaction handling thread for reconnect after a large replay
- After a discussion with @cody-littley, we came to the conclusion that it makes sense to flush transaction handling as part of the replay process, so that a node coming online is fresh and ready to go when replay completes

### PR DESCRIPTION
**Pull request by alittley**
- closes #10923

### CODE REVIEW

**Comment by lpetrovic05** [comment]
Does this fix the JRS test?

**Comment by alittley** [comment]
> Does this fix the JRS test?

@lpetrovic05 Yes. I am still working on getting to the bottom of why exactly the problem is arising. I think this is a good change regardless, but I want to make sure it isn't papering over a problem

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java [hunk 1]
```diff
@@ -831,6 +832,17 @@ public class SwirldsPlatform implements Platform {
...
 
+    /**
+     * Wait until the consensus round handler is not busy.
+     */
+    private void waitUntilTransactionHandlingThreadIsNotBusy() {
+        try {
+            consensusRoundHandler.waitUntilNotBusy();
+        } catch (final InterruptedException e) {
+            throw new RuntimeException("Interrupted waiting for transaction handling thread to not be busy", e);
+        }
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:843]:
```suggestion
        } catch (final InterruptedException e) {
            Thread.currentThread().interrupt();
            throw new RuntimeException("Interrupted waiting for transaction handling thread to not be busy", e);
        }
```

It's best practice to always re-interrupt the current thread if you catch an interrupted exception and rethrow it as something else.

Comment (alittley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java]: Done

**Review by cody-littley — APPROVED** [review]

**Comment by lpetrovic05** [comment]
> > Does this fix the JRS test?
> 
> @lpetrovic05 Yes. I am still working on getting to the bottom of why exactly the problem is arising. I think this is a good change regardless, but I want to make sure it isn't papering over a problem

In that case also close the JRS ticket I opened

**Review by lpetrovic05 — APPROVED** [review]

**Pull request merged**